jax/random.py

```diff
@@ -196,23 +197,45 @@ def _fold_in(key, data):
   return threefry_2x32(key, key2)
 
 
+@partial(jit, static_argnums=(1, 2))
 def _random_bits(key, bit_width, shape):
   """Sample uniform random bits of given width and shape using PRNG key."""
   if not _is_prng_key(key):
     raise TypeError("_random_bits got invalid prng key.")
   if bit_width not in (32, 64):
     raise TypeError("requires 32- or 64-bit field width.")
-  max_count = (bit_width // 32) * onp.prod(shape)
-  if max_count >= onp.iinfo(onp.uint32).max:
-    # TODO(mattjj): just split the key here
-    raise TypeError("requesting more random bits than a single call provides.")
-
-  counts = lax.tie_in(key, lax.iota(onp.uint32, max_count))
-  bits = threefry_2x32(key, counts)
-  if bit_width == 64:
-    bits = [lax.convert_element_type(x, onp.uint64) for x in np.split(bits, 2)]
-    bits = (bits[0] << onp.uint64(32)) | bits[1]
-  return lax.reshape(bits, shape)
+
+  n = (bit_width // 32) * prod(shape)  # number of 32-bit values to generate
+  num_chunks, extra = divmod(n, _max_randvals_per_key)
```

Comment:
Perhaps define a helper function `ceil_of_ratio` in the way that XLA does in C++? Would make it clearer.